File: tutorials/ZioHttpServer/src/main/java/net/officefloor/tutorial/ziohttpserver/MessageRepository.java
package net.officefloor.tutorial.ziohttpserver;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

/**
 * {@link Repository} for {@link Message}.
 * 
 * @author Daniel Sagenschneider
 */
// START SNIPPET: tutorial
@Repository
public interface MessageRepository extends CrudRepository<Message, Integer> {
}
// END SNIPPET: tutorial

File: tutorials/features/polyglot/CatsHttpServer/src/main/scala/net/officefloor/tutorial/catshttpserver/MessageRepository.scala
package net.officefloor.tutorial.catshttpserver

import cats.effect.IO
import doobie._
import doobie.implicits._

/**
 * Message repository.
 */
// START SNIPPET: tutorial
object MessageRepository {

  def findById(id: Int)(implicit xa: Transactor[IO]): IO[Message] =
    sql"SELECT id, content FROM message WHERE id = $id".query[Message].unique.transact(xa)
}
// END SNIPPET: tutorial